table-api/src/main/java/io/deephaven/api/ColumnName.java

```diff
@@ -0,0 +1,61 @@
+package io.deephaven.api;
+
+import java.util.regex.Pattern;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+@Immutable(builder = false, copy = false)
+public abstract class ColumnName
+    implements JoinMatch, JoinAddition, Selectable, Expression, Filter {
+
+    private final static Pattern COLUMN_NAME_PATTERN = Pattern.compile("[a-zA-Z0-9_]+");
+
+    public static boolean isValidColumnName(String name) {
+        return COLUMN_NAME_PATTERN.matcher(name).matches();
+    }
+
+    public static ColumnName of(String name) {
```

Comment:
Anything that calls this might need to take into account leading and trailing whitespace.